runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/bank/Rs2Bank.java

```diff
@@ -1160,31 +1193,41 @@ public static boolean handleBankPin(String pin) {
     }
 
     /**
-     * Banks items if your inventory not enough emptyslots. Will walk back to the initialplayerlocation passed as param
+     * Banks items if your inventory does not have enough emptyslots (0 emptyslots being full). Will walk back to the initialplayerlocation passed as param
      *
      * @param itemNames
      * @param initialPlayerLocation
      * @param emptySlotCount
-     *
      * @return
      */
     public static boolean bankItemsAndWalkBackToOriginalPosition(List<String> itemNames, WorldPoint initialPlayerLocation, int emptySlotCount) {
-        return bankItemsAndWalkBackToOriginalPosition(itemNames, initialPlayerLocation, emptySlotCount, 4);
+        return bankItemsAndWalkBackToOriginalPosition(itemNames, getNearestBank(), initialPlayerLocation, emptySlotCount, 4);
     }
 
     /**
-     * Banks items if your inventory not enough emptyslots. Will walk back to the initialplayerlocation passed as param
+     * Banks items if your inventory is full. Will walk back to the initialplayerlocation passed as param
      *
      * @param itemNames
      * @param initialPlayerLocation
+     * @return
+     */
+    public static boolean bankItemsAndWalkBackToOriginalPosition(List<String> itemNames, WorldPoint initialPlayerLocation) {
+        return bankItemsAndWalkBackToOriginalPosition(itemNames, getNearestBank(), initialPlayerLocation, 0, 4);
+    }
+
+    /**
+     * Banks at specific bank location if your inventory does not have enough emptyslots (0 emptyslots being full). Will walk back to the initialplayerlocation passed as param
+     *
+     * @param itemNames
+     * @param initialPlayerLocation
+     * @param bankLocation
      * @param emptySlotCount
      * @param distance
-     *
      * @return
      */
-    public static boolean bankItemsAndWalkBackToOriginalPosition(List<String> itemNames, WorldPoint initialPlayerLocation, int emptySlotCount, int distance) {
+    public static boolean bankItemsAndWalkBackToOriginalPosition(List<String> itemNames, BankLocation bankLocation, WorldPoint initialPlayerLocation, int emptySlotCount, int distance) {
         if (Rs2Inventory.getEmptySlots() <= emptySlotCount) {
-            boolean isBankOpen = Rs2Bank.walkToBankAndUseBank();
+            boolean isBankOpen = Rs2Bank.walkToBankAndUseBank(bankLocation);
```

Comment:
why are we passing banklocation here? Shouldn't we by default always use the closest available bank?

Reply:
I think the Github change view may not be giving the entire picture here... The adjustments I made around this was to adjust the original methods to allow someone to pass a BankLocation in from BankLocation.java... 

IF you see, I had created more overloaded methods to maintain combability with previous scripts that by default, will go to NearestBank Location.

Reply:
Oh my bad, wasn't clear from the github change view. I understand now.